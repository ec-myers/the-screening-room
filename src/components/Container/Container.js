import React from 'react';
import { connect } from 'react-redux';
import MovieCard from '../MovieCard/MovieCard';
import './Container.scss';

const Container = ({ movies, toggleFavorites }) => {
  console.log(toggleFavorites);
  const allMovies = movies.map(movie => (
    <MovieCard {...movie} key={movie.id} toggleFavorites={toggleFavorites} />
  ));
  return <main className='Container'>{allMovies}</main>;
};

const mapStateToProps = ({ movies }) => ({
  movies,

});

export default connect(mapStateToProps)(Container);
