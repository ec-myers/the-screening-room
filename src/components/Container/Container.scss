.Container {
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(3, 1fr);
  height: 100vh;
  overflow: scroll;
  padding: 35px;
}