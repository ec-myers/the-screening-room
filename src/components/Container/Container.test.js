import React from 'react';
import { shallow } from 'enzyme';
import { Container, mapStateToProps } from './Container';

describe('Container', () => {
  let allMoviesWrapper;

  it('should match the All Movies snapshot', () => {
    let movies = [{ id: 1 }, { id: 2 }, { id: 3 }];
    
    allMoviesWrapper = shallow(<Container movies={movies} />);

    expect(allMoviesWrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with the movies data array', () => {
    const mockState = {
      movies: [{ id: 1 }, { id: 2 }, { id: 3 }]
    };
    const expected = {
      movies: [{ id: 1 }, { id: 2 }, { id: 3 }]
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});
