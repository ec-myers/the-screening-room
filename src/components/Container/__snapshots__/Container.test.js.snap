// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the All Movies snapshot 1`] = `
<main
  className="Container"
>
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 1,
      }
    }
    updateFavorites={[MockFunction]}
  />
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 2,
      }
    }
    updateFavorites={[MockFunction]}
  />
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 3,
      }
    }
    updateFavorites={[MockFunction]}
  />
</main>
`;
